### ISSUE TITLE
Generate @SafeVarargs when necessary

### ISSUE THREAD

**Issue by chrisalice** [google/FreeBuilder#180]
Collection properties can generate varargs `addAllX` methods that require a `@SafeVarargs` annotation for warning-free usage. We should check the method is not overridden and mark it as `@SafeVarargs final` if possible.

**Comment by mdaniel** [comment]
I came to the issues to file a bug about the heap pollution warning, but it seems the project is already aware

So in case others experience the heap pollution warning, if your usage of FreeBuilder tolerates doing so, one can change the builder method to return `Optional<>` and it switches out of the `add(Thing<>... items)` generation mode to just a traditional `set(Collection<Thing<>> value)` mode

For example, change:

```java
interface Polluter<T> { T getT(); }
@FreeBuilder
interface MyData {
  Collection<Polluter<String>> getHeapPolluter();
}
```

to:
```java
interface Polluter<T> { T getT(); }
@FreeBuilder
interface MyData {
  Optional<Collection<Polluter<String>>> getNonPolluting();
}
```

**Comment by chrisalice** [comment]
You can also override the varargs method in the Builder and add the missing annotation

**Comment by chrisalice** [comment]
This has been fixed in [v1.14.2](https://github.com/google/FreeBuilder/releases/tag/v1.14.2).
